components/forms/TransactionSigning.tsx

```diff
@@ -28,10 +32,13 @@ const TransactionSigning = (props: Props) => {
   const [hasSigned, setHasSigned] = useState(false);
   const [walletType, setWalletType] = useState<"Keplr" | "Ledger">();
   const [ledgerSigner, setLedgerSigner] = useState({});
+  const [loading, setLoading] = useState<LoadingStates>({});
 
   const connectKeplr = async () => {
     try {
+      setLoading((newLoading) => ({ ...newLoading, keplr: true }));
```

Comment:
Isn't that the old value in the argument you get there?

```suggestion
      setLoading((oldLoading) => ({ ...oldLoading, keplr: true }));
```